pkg/controllers/status/workstatus_controller.go

```diff
@@ -294,6 +294,16 @@ func (c *WorkStatusController) reflectStatus(work *workv1alpha1.Work, clusterObj
 		return nil
 	}
 
+	var resourceHealth workv1alpha1.ResourceHealth
+	healthy, err := c.ResourceInterpreter.InterpretHealth(clusterObj)
```

Comment:
If you think don't need to check whether `InterpretHealth` is enabled or not, please add a comment on it.

By the way, we can't always return `true` here:
https://github.com/karmada-io/karmada/blob/513f7360e8718efe81701bc4e359f2efe173befd/pkg/resourceinterpreter/defaultinterpreter/default.go#L65-L66

Reply:
Update.